nep-x1.mediawiki

```diff
@@ -0,0 +1,161 @@
+
+==Specification==
+
+NEP-11 compliant contracts MAY implement this NEP for royalties to provide a standard method of specifying royalty payment information.
+
+===Common methods===
+
+====royaltyInfo====
+
+<pre>
+{
+  "name": "royaltyInfo",
+  "safe": true,
+  "parameters": [
+    {
+      "name": "tokenId",
+      "type": "ByteString"
+    },
+    {
+      "name": "royaltyToken",
+      "type": "Hash160"
+    },
+    }
+    {
+      "name": "salePrice",
+      "type": "Integer"
+    }
+  ],
+  "returntype": "Array"
+}
+</pre>
+
+Returns a NeoVM Array stack item with single or multi array, each array still has two elements, for example:
+
+Single royaltyRecipient
+
+<pre>
+[
+    [
+        {
+            "name":"royaltyRecipient",
+            "type":"Hash160"
+        },
+        {
+            "name":"royaltyAmount",
+            "type":"Integer"
+        }
+    ]
+]
+</pre>
+
+Multi royaltyRecipient
+
+<pre>
+[
+    [
+        {
+            "name":"royaltyRecipient1",
+            "type":"Hash160"
+        },
+        {
+            "name":"royaltyAmount1",
+            "type":"Integer"
+        }
+    ],
+    [
+        {
+            "name":"royaltyRecipient2",
+            "type":"Hash160"
+        },
+        {
+            "name":"royaltyAmount2",
+            "type":"Integer"
+        }
+    ],
+    [
+        {
+            "name":"royaltyRecipient3",
+            "type":"Hash160"
+        },
+        {
+            "name":"royaltyAmount3",
+            "type":"Integer"
+        }
+    ]
+]
+</pre>
+
+<code>royaltyToken</code> is the token hash for royalty payment.
+
+<code>salePrice</code> MUST be amount without decimals.
+
+<code>royaltyRecipient</code> is the address of who will be sent the royalty payment, MUST be a 20-byte address.
+
+<code>royaltyAmount</code>  MUST be the royalty amount without decimals rather than a percentage so there's no dispute between marketplaces about how much is the calculated result. 
+
+This function is not for initial sale or mint, it's only for secondary marketplaces.
+
+Marketplaces that support this method MAY implement any method of calculating or transferring royalties to the royalty recipient.
+
+Marketplaces MUST pay the royalty in the same unit of exchange as that of the <code>salePrice</code> passed to royaltyInfo(). For example, if the sale price is in NEO, then the royalty payment must also be paid in NEO, and if the sale price is in GAS, then the royalty payment must also be paid in GAS.
+
+The royaltyInfo() function MUST NOT ignore the <code>salePrice</code>, <code>royaltyAmount</code> MAY be dynamic due to <code>salePrice</code> and time. 
+
+Marketplaces that support this standard MUST emit the event, <code>RoyaltiesTransferred</code>, after sending a payment.
```

Comment:
Marketplaces that support this standard MUST emit the event for each recipient, <code>RoyaltiesTransferred</code>, after sending a payment.